spyder/plugins/editor/widgets/codeeditor.py

```diff
@@ -2039,30 +2017,6 @@ def notify_close(self):
             return params
 
     # -------------------------------------------------------------------------
-    def set_debug_panel(self, show_debug_panel, language):
-        """Enable/disable debug panel."""
-        debugger_panel = self.panels.get(DebuggerPanel)
-        if (is_text_string(language) and
-                language.lower() in ALL_LANGUAGES['Python'] and
-                show_debug_panel):
-            debugger_panel.setVisible(True)
-        else:
-            debugger_panel.setVisible(False)
```

Comment:
This is what I think you should reimplement in `_add_codeeditor` so that `BreakpointsManager` is only added for Python files.